planning/behavior_path_planner/src/scene_module/lane_change/normal.cpp

```diff
@@ -878,180 +909,186 @@ bool NormalLaneChange::getLaneChangePaths(
-      lane_change_info.length = LaneChangePhaseInfo{prepare_length, lane_changing_length};
-      lane_change_info.current_lanes = current_lanes;
-      lane_change_info.target_lanes = target_lanes;
-      lane_change_info.lane_changing_start = prepare_segment.points.back().point.pose;
-      lane_change_info.lane_changing_end = target_segment.points.front().point.pose;
-      lane_change_info.shift_line = utils::lane_change::getLaneChangingShiftLine(
-        prepare_segment, target_segment, target_lane_reference_path, shift_length);
-      lane_change_info.lateral_acceleration = lateral_acc;
-      lane_change_info.terminal_lane_changing_velocity = terminal_lane_changing_velocity;
-
-      const auto candidate_path = utils::lane_change::constructCandidatePath(
-        lane_change_info, prepare_segment, target_segment, target_lane_reference_path,
-        sorted_lane_ids);
-
-      if (!candidate_path) {
-        RCLCPP_DEBUG(logger_, "no candidate path!!");
-        continue;
-      }
+        if (target_lane_reference_path.points.empty()) {
+          RCLCPP_DEBUG(logger_, "target_lane_reference_path is empty!!");
+          continue;
+        }
 
-      const auto is_valid =
-        hasEnoughLength(*candidate_path, current_lanes, target_lanes, direction);
+        LaneChangeInfo lane_change_info;
+        lane_change_info.longitudinal_acceleration =
+          LaneChangePhaseInfo{longitudinal_acc_on_prepare, longitudinal_acc_on_lane_changing};
+        lane_change_info.duration = LaneChangePhaseInfo{prepare_duration, lane_changing_time};
+        lane_change_info.velocity =
+          LaneChangePhaseInfo{prepare_velocity, initial_lane_changing_velocity};
+        lane_change_info.length = LaneChangePhaseInfo{prepare_length, lane_changing_length};
+        lane_change_info.current_lanes = current_lanes;
+        lane_change_info.target_lanes = target_lanes;
+        lane_change_info.lane_changing_start = prepare_segment.points.back().point.pose;
+        lane_change_info.lane_changing_end = target_segment.points.front().point.pose;
+        lane_change_info.shift_line = utils::lane_change::getLaneChangingShiftLine(
+          prepare_segment, target_segment, target_lane_reference_path, shift_length);
+        lane_change_info.lateral_acceleration = lateral_acc;
+        lane_change_info.terminal_lane_changing_velocity = terminal_lane_changing_velocity;
+
+        const auto candidate_path = utils::lane_change::constructCandidatePath(
+          lane_change_info, prepare_segment, target_segment, target_lane_reference_path,
+          sorted_lane_ids);
+
+        if (!candidate_path) {
+          RCLCPP_DEBUG(logger_, "no candidate path!!");
+          continue;
+        }
 
-      if (!is_valid) {
-        RCLCPP_DEBUG(logger_, "invalid candidate path!!");
-        continue;
-      }
+        const auto is_valid =
+          hasEnoughLength(*candidate_path, current_lanes, target_lanes, direction);
```

Comment:
I thought the feature is better to be implemented in the differentt function because the feature is not checking whether the path is long enough or not.

Reply:
done
https://github.com/autowarefoundation/autoware.universe/pull/5068/commits/cfd377fac81f0c4deebd3625b16116153dcc7285